tools/dependency/release_dates.py

```diff
@@ -38,35 +40,67 @@ def format_utc_date(date):
 def verify_and_print_latest_release(dep, repo, metadata_version, release_date):
     try:
         latest_release = repo.get_latest_release()
-        if latest_release.created_at > release_date and latest_release.tag_name != metadata_version:
-            print(
-                f'*WARNING* {dep} has a newer release than {metadata_version}@<{release_date}>: '
-                f'{latest_release.tag_name}@<{latest_release.created_at}>')
-    except github.UnknownObjectException:
-        pass
+    except github.GithubException as err:
+        # Repositories can not have releases or if they have releases may not publish a latest releases. Return
+        print(f'GithubException {repo.name}: {err.data} {err.status} while getting latest release.')
+        return
+    if latest_release.created_at > release_date and latest_release.tag_name != metadata_version:
+        print(
+            f'{Fore.YELLOW}*WARNING* {dep} has a newer release than {metadata_version}@<{release_date}>: '
+            f'{latest_release.tag_name}@<{latest_release.created_at}>{Style.RESET_ALL}')
 
 
-# Print GitHub release date, throw ReleaseDateError on mismatch with metadata release date.
+# Print GitHub release date, throw ReleaseDateVersionError on mismatch with metadata release date.
 def verify_and_print_release_date(dep, github_release_date, metadata_release_date):
     mismatch = ''
     iso_release_date = format_utc_date(github_release_date)
-    print(f'{dep} has a GitHub release date {iso_release_date}')
+    print(f'{Fore.GREEN}{dep} has a GitHub release date {iso_release_date}{Style.RESET_ALL}')
```

Comment:
The way this is structured today makes finding the date mismatches easy to find. The script outputs the real GH date followed by the exception that shows the erroneous date. I don't expect many people to view this but helpful for myself and the deps maintainers. But let me know if I'm missing something

Reply:
kinda unreliable iiuc - because one is going to stdout and the other to stderr

there is no guarantee that both will be shown and if they are that they will be in correct order

not a blocker - but that is the reason i suggested this

Reply:
(also that it flashes green even when its about to fail)